pyproject.toml

```diff
@@ -0,0 +1,150 @@
+    {name = "Winston Chang", email = "winston@posit.co"},
+]
+description = "A web development framework for Python."
+readme = "README.md"
+requires-python = ">=3.8"
+license = {text = "MIT"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+    "typing-extensions>=4.10.0",
+    "uvicorn>=0.16.0;platform_system!='Emscripten'",
+    "starlette",
+    "websockets>=10.0",
+    "python-multipart",
+    "htmltools>=0.5.2",
+    "click>=8.1.4;platform_system!='Emscripten'",
+    "markdown-it-py>=1.1.0",
+    "mdit-py-plugins>=0.3.0",
+    "linkify-it-py>=1.0",
+    "appdirs>=1.4.4",
+    "asgiref>=3.5.2",
+    "packaging>=20.9",
+    "watchfiles>=0.18.0;platform_system!='Emscripten'",
+    "questionary>=2.0.0;platform_system!='Emscripten'",
+    "prompt-toolkit;platform_system!='Emscripten'",
+    "python-multipart>=0.0.7;platform_system!='Emscripten'",
+    "setuptools;python_version>='3.12'",
+]
+
+[project.optional-dependencies]
+theme = ["libsass>=0.23.0"]
+test = [
+    "pytest>=6.2.4",
+    "pytest-asyncio>=0.17.2",
+    "pytest-playwright>=0.3.0",
+    "playwright>=1.43.0",
+    "pytest-xdist",
+    "pytest-timeout",
+    "pytest-rerunfailures",
+    "pytest-cov",
+    "coverage",
+    "syrupy",
+    "psutil",
+    "astropy",
+    "suntime",
+    "timezonefinder",
+    "ipyleaflet",
+    "shinywidgets",
+    "seaborn",
+    "plotnine",
+    "plotly",
+    "duckdb",
+    "holoviews",
+    "bokeh",
+    "xarray",
+    "geopandas",
+    "geodatasets",
+    "missingno",
+    "rsconnect-python",
+    "scikit-learn",
+    "folium",
+    "palmerpenguins",
+    "faicons",
+    "ridgeplot",
+    "great_tables",
+]
+dev = [
+    "black>=24.0",
+    "flake8>=6.0.0",
+    "flake8-bugbear>=23.2.13",
+    "Flake8-pyproject>=1.2.3",
+    "isort>=5.10.1",
+    "libsass>=0.23.0",
+    "pyright==1.1.369",
+    "pre-commit>=2.15.0",
+    "wheel",
+    "matplotlib",
+    "pandas",
+    "pandas-stubs",
+    "polars",
+    "numpy",
+    "shinyswatch>=0.7.0",
+    "python-dotenv",
+    "anthropic",
+    "google-generativeai;python_version>='3.9'",
+    "langchain_core",
+    "openai",
+    "ollama",
+    "tokenizers",
+    "aiohttp",
+    "beautifulsoup4",
+]
+doc = [
+    "jupyter",
+    "jupyter_client < 8.0.0",
+    "tabulate",
+    "shinylive",
+    "pydantic>=2.7.4",
+    "quartodoc==0.7.5",
+    "griffe<1.0.0",
+]
+
+[project.urls]
+Homepage = "https://github.com/posit-dev/py-shiny"
+Documentation = "https://shiny.posit.co/py/"
+Repository = "https://github.com/posit-dev/py-shiny"
+"Bug Tracker" = "https://github.com/posit-dev/py-shiny/issues"
+
+[build-system]
```

Comment:
As a developer, it feels helpful to have build-system and tools.setuptools at the top, since it hints at what sections to expect in pyproject.toml / how things get made

Reply:
A helpful example might be ibis. They put their poetry config up top and build system at bottom. But because poetry is at the top you know what the build system will be set to.

Reply:
Thanks!